### DISCUSSION TITLE
[4.1] User Group Page Classes

### DISCUSSION THREAD

**Discussion by joomleb** [joomla/joomla-cms#34796] [2. Cold Features]
Hi guys, 
to have add a **Page Class** depending on the User (Guest or Logged) **User Group** and _following the the parents scheme_ would be too useful for example for styling the layouts, hiding fields, disable buttons {pointer-events: none;} etc. 
One more big weapon for User Group, ACL and Permissions features.

I'm not a real developer, but the solution could be quite simple by adding a "Class" filed into the User Group Details (or directly by using the User Group Title) to use as as Page Class name to add.

Approximately something like (sorry for all the mistakes):
```
<?php $user =& JFactory::getUser(); 
    $pageclass = ""; 
    if(array_key_exists('12',$user->groups)){ 
    $pageclass = "regd"; 
    } elseif(array_key_exists('11',$user->groups)){ 
        $pageclass = "regc"; 
        } elseif(array_key_exists('10',$user->groups)){ 
            $pageclass = "regb"; 
            } elseif(array_key_exists('2',$user->groups)){ 
                $pageclass = "rega"; 
                } elseif(array_key_exists('9',$user->groups)){ 
                    $pageclass = "guest"; 
                }
    if(array_key_exists('6',$user->groups)){ 
    $pageclass = "manager"; 
    } 
    if(array_key_exists('8',$user->groups)){ 
    $pageclass = "superuser"; 
    } 
?>
```

**Comment by brianteeman** [comment]
What happens if a user is in more than one usergroup?

**Comment by joomleb** [comment]
Hi @brianteeman, 
In first of all, sure, will be good a selector to say "consider the User Group or no" for the Page Class writing (so users could activate it just for the User Groups needed). 
And, others rules / controls can be added to prevent manual errors etc. But for now I avoid them, I don't mention them here, to directly answer to you in a simpler way...  

it should check "if in the User Group" by checking from the highest level for each chain of parents User Groups. So:
- if in more than one User Group on the same chain parents User Group, only the highest level will be added
- If in  another User Groups chain, also this User Group class will be added (different User Groups parents chains could / should mean different scopes) 

**For example**, I have two chains of parents User Groups:
Guest
- Registered
-- Registered B
--- Registered C
---- Registered D
On this first chain I will select all of them to consider for the Page Class writing
Guest
- Registered
-- Author
--- Editor
---- Publisher
On this second chain I will select to consider just Publisher > Editor > Author for the Page Class writing.

**Results Examples**:
- If the User is not Logged, "guest" will be added to the Page Class
- If the User is Logged and is added to all User Groups, "registeredd" and "publisher" will be added to the Page Class
- If the User is Logged and is added to the "Registered" and "Registered C" User Groups, "registeredc" will be added to the Page Class
